### ISSUE TITLE
multi-class AsymmetricUnifiedFocalLoss

### ISSUE THREAD

**Issue by tomanick** [oikosohn/compound-loss-pytorch#1]
Thanks for your sharing!
If I want to use multi-class AsymmetricUnifiedFocalLoss, is current version available?

**Comment by oikosohn** [comment]
I think you can. Because official code also supports multi-class. I just converted the official repository code into a PyTorch. But I did not test. so If you can't, please let me know.

**Comment by Sean-asc** [comment]
@tomanick  Does AsymmetricUnifiedFocalLoss works fine on multi-class? 
                    I tried, but got some unexpected errors.

**Comment by Sean-asc** [comment]
# Calculate losses separately for each class, only enhancing foreground class
        back_dice = (1-dice_class[:,0]) 
        fore_dice = (1-dice_class[:,1]) * torch.pow(1-dice_class[:,1], -self.gamma) 

This is not suitable for multi-class, right?

**Comment by aliwaqas333** [comment]
> # Calculate losses separately for each class, only enhancing foreground class
> ```
>     back_dice = (1-dice_class[:,0]) 
>     fore_dice = (1-dice_class[:,1]) * torch.pow(1-dice_class[:,1], -self.gamma) 
> ```
> 
> This is not suitable for multi-class, right?

Yes this assumes one channel is background and another is foreground.

**Comment by AlbertTJU** [comment]
> > # Calculate losses separately for each class, only enhancing foreground class
> > ```
> >     back_dice = (1-dice_class[:,0]) 
> >     fore_dice = (1-dice_class[:,1]) * torch.pow(1-dice_class[:,1], -self.gamma) 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This is not suitable for multi-class, right?
> 
> Yes this assumes one channel is background and another is foreground.

Hello, my output tensor and label tensor only have one channel, so how to modify this code?

**Comment by dyollb** [comment]
There is an issue with a snippet in the official tensor flow repo: https://github.com/mlyg/unified-focal-loss/issues/8
